hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -270,7 +269,8 @@ private void uninitialize() {
     metrics.clear();
     if (replicationEndpoint.state() == Service.State.STARTING
         || replicationEndpoint.state() == Service.State.RUNNING) {
-      replicationEndpoint.stopAndWait();
+      replicationEndpoint.stopAsync();
```

Comment:
This unit contains several changes required by evolution in Guava's Service interface. Our ReplicationEndpoint extends this interface.